### ISSUE TITLE
[BUG] Result values are in wrong columns when executing query

### ISSUE THREAD

**Issue by techfg** [cweijan/vscode-mysql#1266]
- Database type and version:
Postgres 15.4

When executing certain queries, there are values displayed in the wrong columns.  For example, in the query below, column `4` has a value of `1500`, column `5` contains `1000` and column `6` contains `1200` when they should all be NULL as the first column with a value should be `7`.  Note that the results displayed in column `7` thru `12` are correct.

```sql
DROP TABLE IF EXISTS tmp_sales;
CREATE TEMP TABLE tmp_sales(year int, month int, field1 text, field2 text, field3 text, qty int);
INSERT INTO tmp_sales values(2007, 7, 'foo', 'bar', 'fiddly', 1500);
INSERT INTO tmp_sales values(2007, 8, 'foo', 'bar', 'fiddly', 1000);
INSERT INTO tmp_sales values(2007, 9, 'foo', 'bar', 'fiddly', 1200);
INSERT INTO tmp_sales values(2007, 10, 'foo', 'bar', 'fiddly', 500);
INSERT INTO tmp_sales values(2007, 10, 'foo', 'bar', 'fiddly', 1500);
INSERT INTO tmp_sales values(2007, 10, 'foo', 'bar', 'fiddly', 2000);
INSERT INTO tmp_sales values(2008, 11, 'foo', 'bar', 'fiddly', 1000);
SELECT * FROM crosstab(
    'SELECT year, field1, field2, field3, month, qty FROM tmp_sales ORDER BY year', 
    'SELECT m FROM generate_series(1, 12, 1) m'
) t(year int, field1 text, field2 text, field3 text, "1" text, "2" text, "3" text, "4" text, "5" text, "6" text, "7" text, "8" text, "9" text, "10" text, "11" text, "12" text)
```

**Steps to reproduce**
1. Ensure that [tablefunc](https://www.postgresql.org/docs/current/tablefunc.html) is installed
2. Execute query block from above

**Actual Result**
Columns `4/5/6` contain the values `1500`/`1000`/`1200` and `7/8/9/10/11/12` contain the correct values

**Expected Result**
Columns `4/5/6` should contain `NULL`

**Additional Information**
1. If you reduce the number of "extra" fields (e.g., `field1`, `field2`, `field3`) the results shift to the right by the number of "extra" columns.  For example, if you only have `field1` and `field2`, only columns `5/6` contain incorrect data.  If you have just `field1`, only column `6` will have incorrect data and if you do not have any extra fields, all the results are correct.
2. When executing the same statements in pgAdmin, the results are correctly displayed

Database Client Result
![image](https://github.com/user-attachments/assets/93e8140c-c5a3-4242-9fd2-0ec6cf112742)

pgAdmin Result
![image](https://github.com/user-attachments/assets/801d04d8-71d7-4bfa-a67c-6706c33c1833)

**Comment by cweijan** [comment]
Thanks for the feedback, this is because the column name is a number and is incorrectly parsed as an array index, which has been fixed in version 7.6.1.

**Comment by techfg** [comment]
@cweijan - Confirmed that results display correctly in 7.6.1, thanks for the quick fix!
